core/device.cpp

```diff
@@ -44,6 +44,21 @@ extern "C" const struct device *get_device_for_dc(const struct device_table *tab
 	return it != dcs.end() && same_device(*it, dev) ? &*it : NULL;
 }
 
+extern "C" const struct device *get_or_add_device_for_dc(struct device_table *table, const struct divecomputer *dc, int *idx)
```

Comment:
I'm not strictly against this interface, but it would appear more reasonable to me to just return and index (-1 for not found) and then use `get_device(...)` by the caller if they need to.

Something like the completely untested:
```
extern "C" int get_or_add_device_for_dc(struct device_table *table, const struct divecomputer *dc)
{
        if (!dc->model || !dc->serial || !idx)
                return -1;

        device dev { dc->model, dc->serial };
        auto it = std::lower_bound(table->devices.begin(), table->devices.end(), dev);
        if (it != dcs.end() && same_device(*it, dev))
                return it - table->devices.begin();
        else
                return create_device_node(table, dc->model, dc->serial, "");
}
```

Reply:
yeah, I guess I could do that - it's only one caller so far, anyway.
I'll play with it.

Reply:
that was a GREAT suggestion. The resulting code is actually MUCH nicer. Well done.